### ISSUE TITLE
option except

### ISSUE THREAD

**Issue by bgbruno** [sindresorhus/kill-tabs#19]
**type**
idea

**issue**
I need many tabs opened and use only time to time - meanwhile they decreace performance - so I want kill just them, and my important still have active

**solution**
create command-line option "except"
• except-domain -- eg `$ kill-tabs --except-domain google.com`
• except-title -- eg `$ kill-tabs --except-title google`

**Comment by GeorgeGkas** [comment]
I investigated this case for Windows and seems it is not currently possible to retrieve the window title of a browser process  the from command line. Otherwise, the simplest approach would be to get the output of the following command for each of the processes returned by `wmic` in `windows.js`:

```powershell
Get-Process -id $ID |format-table mainwindowtitle -Auto
```

where `$ID` is the process id we retrieved. However this will not work. No browser (Edge, Chrome, Brave, Chronium) export the name of the current active window in its relevant process info.

I don't know if this is the case for other OSs, but from what I concluded, this cannot be achieved on Windows. Maybe using the Windows API in a custom C++ program exported as Node package, but then again, I don't think it's worth the trouble, or even if it's even possible.

**Comment by bgbruno** [comment]
yes I did investigation too 2 weeks ago for #macos - and seems it is not possible - for now

maybe chrome have some "command line swither" - but if have it is not default solution

so only solution what I saw was "own chrome extension"

**Comment by kitschpatrol** [comment]
Couple years late, but you can filter and close tabs by domain in most browsers on macOS with AppleScript:

Hard-coded `--except-domain google.com` example:

```applescript
tell application "Google Chrome"
  set windowList to every window
  repeat with win in windowList
    set tabList to every tab of win
    repeat with t in tabList
      if not (URL of t contains "google.com") then
        close t
      end if
    end repeat
  end repeat
end tell
```

You can invoke AppleScript from Node via a `child_process` call to `osascript -e`.

**Comment by sindresorhus** [comment]
This is out of scope.
